presentation/src/main/java/me/androidbox/presentation/event/screen/EventScreen.kt

```diff
@@ -78,14 +90,24 @@ fun EventScreen(
 
                     Spacer(modifier = modifier.height(26.dp))
                     AgendaDuration(
-                        startTime = "08:00",
-                        endTime = "08:30",
-                        startDate = "Jul 21 2022",
-                        endDate = "Jul 21 2022",
-                        modifier = Modifier.fillMaxWidth()
+                        isEditMode = true,
+                        startTime = eventScreenState.startTimeDuration.formatDateTime(TIME_PATTERN),
+                        endTime = eventScreenState.endTimeDuration.formatDateTime(TIME_PATTERN),
+                        startDate = eventScreenState.startDateDuration.formatDateTime(DATE_PATTERN),
+                        endDate = eventScreenState.endDateDuration.formatDateTime(DATE_PATTERN),
+                        modifier = Modifier.fillMaxWidth(),
+                        onStartDurationClicked = {
+                            isStartDateTime = true
+                            calendarStateTimeDate.show()
+                        },
+                        onEndDurationClicked = {
+                            isStartDateTime = false
+                            calendarStateTimeDate.show()
+                        }
```

Comment:
I'd completely remove the `isStartDateTime` boolean for several reasons:
1. As mentioned above, it's a side-effect and will lead to undefined behavior
2. Whether your date picker dialog is shown or not (and also which one) is considered state. That boolean should therefore go in your screen state class and the ViewModel should toggle it. That way it'll also survive config changes such as toggling the device theme.

Reply:
@philipplackner Updated and moved this to the EventScreenState